g2p_documents/models/document_file.py

```
@@ -42,14 +41,27 @@ def filter_for_tags_any(self, tags):
             ]
         return self.filtered(lambda x: any((x.tags_ids and tag in x.tags_ids.name) for tag in tags))
 
-    @api.depends("name", "data", "backend_id")
-    def _compute_mime_type(self):
-        for rec in self:
-            if rec.backend_id.mimetype_strategy == "from_file_name" and rec.name:
-                rec.mimetype, __ = mimetypes.guess_type(rec.name)
-            elif rec.data:
-                # Defaults of "from_data" mode even if empty.
-                rec.mimetype = guess_mimetype(base64.b64decode(rec.data))
```

Why is mimetype_strategy based detection removed here?

---

The name field value changes to a random string before reaching _compute_mime_type method so this strategy(from_file_name) won't work.

---

Where is name field value changing to random string before reaching compute? 
And this is intentional from_file_name strategy will only work for document stores where file names are not random. (Even though name will default to random UUID, the user can still change the name on UI.)
(The idea is to give configurability to infer mimetype from filename because always inferring mimetype from data might be costly operation)

---

_compute_mime_type method is triggering multiple time when we manually gave name like aadhar.jpeg
first time it will compute mimetype - jpeg, but then name will change to random string and again this method will be called and this time it will set mimetype to null

aadhar.jpeg
image/jpeg

45b2f7b7-e928-4ac6-92b9-b96edc9536f6
False